### ISSUE TITLE
AttributeError: 'ResultMeta' object has no attribute 'index' with inner hits

### ISSUE THREAD

**Issue by pkphlam** [elastic/elasticsearch-dsl-py#431]
Hi,

I'm getting an AttributeError: 'ResultMeta' object has no attribute 'index' when trying to query with inner_hits with a nested object. It seems to be inconsistently breaking (sometimes I successfully get a hit or so before it breaks). I'm wondering if I'm defining something incorrectly?

```
es.indices.get_mapping('index_nested_test')
{'index_nested_test': {'mappings': {'documents': {'properties': {'text': {'type': 'string'},
     'tokens': {'properties': {'tokens': {'type': 'string'}},
      'type': 'nested'}}}}}}
```

```
s = Search(using=es).index('index_nested_test').doc_type('documents').query('nested', path='tokens', query=Q('term', tokens__tokens='tissu'), inner_hits={})
print(s.count())
5
```

```
for h in s.execute():
    print(h.meta.inner_hits)

{'tokens': <Response: [<Result(index_nested_test/documents/5...}
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/store/notebook/venv/lib/python3.4/site-packages/elasticsearch_dsl/utils.py in __getattr__(self, attr_name)
    119         try:
--> 120             return _wrap(self._d_[attr_name])
    121         except KeyError:

KeyError: 'index'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
<ipython-input-78-777f71561898> in <module>()
      1 for h in s.execute():
----> 2     print(h.meta.inner_hits)

/store/notebook/venv/lib/python3.4/site-packages/elasticsearch_dsl/utils.py in __repr__(self)
    105 
    106     def __repr__(self):
--> 107         r = repr(self._d_)
    108         if len(r) > 60:
    109             r = r[:60] + '...}'

/store/notebook/venv/lib/python3.4/site-packages/elasticsearch_dsl/result.py in __repr__(self)
     24 
     25     def __repr__(self):
---> 26         return '<Response: %r>' % self.hits
     27 
     28     def __len__(self):

/store/notebook/venv/lib/python3.4/site-packages/elasticsearch_dsl/utils.py in __repr__(self)
     34 
     35     def __repr__(self):
---> 36         return repr(self._l_)
     37 
     38     def __eq__(self, other):

/store/notebook/venv/lib/python3.4/site-packages/elasticsearch_dsl/result.py in __repr__(self)
     82     def __repr__(self):
     83         return u('<Result(%s/%s/%s): %s>') % (
---> 84             self.meta.index, self.meta.doc_type, self.meta.id, super(Result, self).__repr__())

/store/notebook/venv/lib/python3.4/site-packages/elasticsearch_dsl/utils.py in __getattr__(self, attr_name)
    121         except KeyError:
    122             raise AttributeError(
--> 123                 '%r object has no attribute %r' % (self.__class__.__name__, attr_name))
    124 
    125     def __delattr__(self, attr_name):

AttributeError: 'ResultMeta' object has no attribute 'index'
```

**Comment by pkphlam** [comment]
I should also note that running the same query through the lower-level python elasticsearch package works perfectly.

This works perfectly fine:
```
es.search('index_nested_test, body=s.to_dict())
```

**Comment by HonzaKral** [comment]
Thanks for the report. It looks like the error is actually just in the `__repr__` method which relies on the metadata like `_index` and `_type` to be present. I pushed a fix for it in 32b119b77fe06eeaaaac84d06960384623763619

I haven't had the chance yet to verify if that also helps with `nested` as opposed to parent/child. Will try asap

**Comment by HonzaKral** [comment]
This has been resolved, thanks again for the report.
